cql/src/main/java/org/apache/cassandra/stargate/transport/internal/Server.java

```diff
@@ -321,12 +323,25 @@ int countConnectedClients() {
         if (connection instanceof ServerConnection) {
           ServerConnection conn = (ServerConnection) connection;
           Optional<AuthenticatedUser> user = conn.persistenceConnection().loggedUser();
-          String name = user.map(AuthenticatedUser::name).orElse(null);
+          String name = user.map(AuthenticatedUser::name).orElse("unknown");
           result.put(name, result.getOrDefault(name, 0) + 1);
         }
       }
       return result;
     }
+
+    Map<ConnectionMetrics, Integer> countConnectedClientsByConnectionMetrics() {
+      Map<ConnectionMetrics, Integer> result = new HashMap<>();
+      for (Channel c : allChannels) {
+        Connection connection = c.attr(Connection.attributeKey).get();
+        if (connection instanceof ServerConnection) {
+          ServerConnection conn = (ServerConnection) connection;
+          ConnectionMetrics connectionMetrics = conn.getConnectionMetrics();
+          result.compute(connectionMetrics, (ci, v) -> v == null ? 1 : v + 1);
```

Comment:
Sorry, I might have missed the comparison logic, but it is possible to have more that one value per key here?